### PR TITLE
importinto: check column assignment expressions in `import into ...`

### LINKED ISSUE #55497: IMPORT INTO should explicitly return error for unsupported use case
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

IMPORT INTO is a complex statement, it may be executed in multiple TiDB nodes in the cluster. And its internal implementation is different from LOAD DATA which users may think they are similar. So we should explicitly return error for unsupported use case.

For example,

```
IMPORT INTO t(a, @, c) SET b=(SELECT 'subquery') FROM
```

will not return error and will not import any data due to an internal panic

### 2. What did you expect to see? (Required)

error message or data is imported.

### 3. What did you see instead (Required)

see above

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #55497

### What changed and how does it work?

This PR does a complete check for column assignments in `IMPORT INTO ...`, it:

- Disallow subquery, column reference, system vars access... in planner pahse.
- Because import into is running across node, it uses a ctx that provides restricted fields. So we also check the `RequiredOptionalEvalProps`  of the expressions to match the context.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/importintotest/import_into_test.go [hunk 1]
```diff
@@ -146,14 +146,19 @@ func (s *mockGCSSuite) TestBasicImportInto() {
...
 	s.tk.MustExec("create table t (a bigint, b varchar(100), c int);")
 	s.tk.MustExec("set @v='test'")
-	sql := fmt.Sprintf(`IMPORT INTO t(a, @, c) SET b=tidb_is_ddl_owner() FROM 'gs://test-multi-load/db.tbl.001.csv?endpoint=%s'
+	// IMPORT INTO does not support column reference
+	sql := fmt.Sprintf(`IMPORT INTO t(a, @, c) SET b=a+1 FROM 'gs://test-multi-load/db.tbl.001.csv?endpoint=%s'
 		with thread=1`, gcsEndpoint)
-	err := s.tk.QueryToErr(sql)
-	require.ErrorContains(s.T(), err, "[0]can't use function at SET index 0")
+	err := s.tk.ExecToErr(sql)
+	require.ErrorContains(s.T(), err, "COLUMN reference is not supported in IMPORT INTO column assignment, index 0")
```
Comment (lance6716) [inline tests/realtikvtest/importintotest/import_into_test.go:156]: This should be supported

**Review by lance6716 — COMMENTED** [review]
/cc @D3Hunter

File: tests/realtikvtest/importintotest/import_into_test.go (hunk 1 above)
Comment (D3Hunter) [reply tests/realtikvtest/importintotest/import_into_test.go:156]: not now. I tried `load data`, mysql support it, but we don't

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]
rest lgtm

**Review by winoros — APPROVED** [review]

**Pull request merged**